### PR TITLE
Add trust toggle / DisableTrustedLocationWarnings pref

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

This PR is based on: https://github.com/DynamoDS/Dynamo/pull/12847

* adds DisableTrustedLocationWarnings pref
* serializes automatically and deserializes manually.
* tries to generalize the work @pinzart90 did for manual deserialization of prefs.
* starts hooking up UI to pref - still in progress.
Screen Shot 2022-05-09 at 1 16 18 PM
Screen Shot 2022-05-09 at 1 15 43 PM

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

Adds preference for disabling trust warnings, this preference is not yet used.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs [hunk 1]
```diff
@@ -373,6 +373,24 @@ public bool DisableCustomPackages
...
+                return preferenceSettings.DisableTrustWarnings;
+            }
+            // We keep this setter private to avoid view extensions calling it directly.
+            // Access modifiers are not intended for security, but it's simple enough to hook a toggle to the UI
+            // without binding, and this makes it clear it's not an API.
+            internal set
+            {
+                preferenceSettings.SetTrustWarningsDisabled(value);
+            }
+        }
```
Comment (pinzart90) [inline src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:393]: I guess the Api is a bit fragmented.
The trusted locations should be used from the trustedLocManager but the TrustedWarnings flag can be used directly from the preferences.
Should they all be in a single place?

Comment (pinzart90) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:393]: The only way I see to make it less confusing is to put all the trusted APIs in the Preferences class. No more manager...
I know in our discussions it seemed like manager had its benefits..but the current way is kinda confusing

Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:393]: I also added a wrapper for retrieving the state of this toggle to the trustedLocManager, is that what you mean?
https://github.com/DynamoDS/Dynamo/pull/12856/files#diff-444ffe085afa9b5863e42b8663f6fead6f785c65f671070e8219897d890225daR162

Comment (pinzart90) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:393]: Both trusted locations API and the warnings flag API can be accessed from both the manager and the preferences class. Can be confusing to decide what to choose,  More so for internal users(developers)

Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs:393]: ok - should I attempt to remove the trust manager class in this PR?

**Review by pinzart90 — APPROVED** [review]

**Comment by QilongTang** [comment]
> tests passed here: https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/6023/parameters/ merging.
> 
> merging.
> 
> @QilongTang @Jingyi-Wen - I think when the final UI gets hooked up here it's worth rethinking the label for this toggle as well as the tooltip text when hovering over the `?`
> 
> 1. `Disable receiving trust warning` - couldn't we simplify this to `disable trust warnings`? It just sounds wordy.
> 2. `We recommend keeping this toggle off to decrease security risks` - as the trusted locations feature doesn't directly increase security in any way - it just warns the user, IMO the messaging is a bit misleading. My suggestion is something like: `We recommend keeping this toggle off to avoid inadvertently open a file from an untrusted location.`

@Jingyi-Wen Let me know what do you think of this suggestion

**Pull request merged**